tests/writer_test.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+"""Tests for the artifact definitions readers."""
+
+import io
+import os
+import unittest
+import json
+
+from artifacts import reader
+from artifacts import writer
+import artifacts.py2to3 as py2to3
+
+
+class YamlArtifactsWriterTest(unittest.TestCase):
+  """Class to test the YAML artifacts writer"""
+
+  def testYamlWriter(self):
+    """Tests the YamlArtifactsWriter FormatArtifacts method for loss during conversion."""
+    artifact_reader = reader.YamlArtifactsReader()
+    artifact_writer = writer.YamlArtifactsWriter()
+    test_file = os.path.join('test_data', 'definitions.yaml')
+
+    artifact_definitions = list(artifact_reader.ReadFile(test_file))
+    artifacts_yaml = artifact_writer.FormatArtifacts(artifact_definitions)
```

Comment:
So I pushed a change to remove the py2to3 stuff in favor of StringIO/BytesIO if you would prefer to merge it that way, but I will probably switch this to actually doing the temp file if that is acceptable.

I only did all of this juggling because I assumed it would be better to not be writing/deleting temp files as part of unit tests, but if that isn't a concern then I can do it that way.